src/MainWindow.vala

```diff
@@ -450,6 +498,7 @@ public class PantheonCalculator.MainWindow : Gtk.ApplicationWindow {
         };
         main_grid.append (basic_grid);
         main_grid.append (extended_revealer);
+        ((Gtk.Widget) main_grid).add_controller (event_controller);
```

Comment:
You should add event controller to the window itself, not main grid.

Reply:
it's late tomorrow